### ISSUE TITLE
translated traffic targets don't work before istio-1.4 due to istio bug

### ISSUE THREAD

**Issue by rickducott** [servicemeshinterface/smi-adapter-istio#70]
Trying to follow the example here: https://github.com/deislabs/smi-adapter-istio/tree/master/docs/smi-traffictarget

I have deployed istio with SMI and an mtls MeshPolicy. I set up ingress and to start I have no destination rules, service roles, service role bindings, http route groups, traffic targets, etc. 

![image](https://user-images.githubusercontent.com/27434432/67884609-80331c00-fb1c-11e9-82bf-46b9c54dcb93.png)

Now I apply the ClusterRbacConfig to set up deny rules everywhere:

```
apiVersion: rbac.istio.io/v1alpha1
kind: ClusterRbacConfig
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"rbac.istio.io/v1alpha1","kind":"ClusterRbacConfig","metadata":{"annotations":{},"name":"default"},"spec":{"mode":"ON"}}
  creationTimestamp: 2019-10-30T17:52:51Z
  generation: 1
  name: default
  resourceVersion: "23019"
  selfLink: /apis/rbac.istio.io/v1alpha1/clusterrbacconfigs/default
  uid: 17dac909-fb3e-11e9-867e-42010a800172
spec:
  mode: "ON"
```

![image](https://user-images.githubusercontent.com/27434432/67884668-9d67ea80-fb1c-11e9-8d4e-eeb77c5e1b34.png)

Now I'll create the productpage viewer service role and binding to make productpage load again, with no data from details, reviews, or services:

```
apiVersion: "rbac.istio.io/v1alpha1"
kind: ServiceRole
metadata:
  name: productpage-viewer
  namespace: bookinfo
spec:
  rules:
    - services: ["productpage.bookinfo.svc.cluster.local"]
      methods: ["GET"]
---
apiVersion: "rbac.istio.io/v1alpha1"
kind: ServiceRoleBinding
metadata:
  name: bind-productpage-viewer
  namespace: bookinfo
spec:
  subjects:
    - user: "*"
  roleRef:
    kind: ServiceRole
    name: "productpage-viewer"
```

So far so good:
![image](https://user-images.githubusercontent.com/27434432/67884802-d2743d00-fb1c-11e9-9545-f340af47deea.png)

Now I want to enable a portion of the application with SMI. So I create a single http route group and traffic target to enable product page to talk to the details service and populate the left hand side here:

```
---
apiVersion: specs.smi-spec.io/v1alpha1
kind: HTTPRouteGroup
metadata:
  name: bookinfo-allowed-paths
  namespace: bookinfo
matches:
- name: api
  pathRegex: ""
  methods: ["*"]
---
kind: TrafficTarget
apiVersion: access.smi-spec.io/v1alpha1
metadata:
  name: productpage-v1-details-v1
  namespace: bookinfo
destination:
  kind: ServiceAccount
  name: details-v1
  namespace: bookinfo
specs:
  - kind: HTTPRouteGroup
    name: bookinfo-allowed-paths
    matches:
      - api
sources:
  - kind: ServiceAccount
    name: productpage-v1
    namespace: bookinfo
```

Now I expect productpage to be able to load details, but fail to load ratings or reviews. However, it appears the reviews are loaded in addition to details, and only the ratings service is unavailable:

![image](https://user-images.githubusercontent.com/27434432/67884968-21ba6d80-fb1d-11e9-8391-1bb3f1fc9538.png)

I looked and confirmed only one service role and role binding were added:
```
[13:55:49] rick@Ricks-MacBook-Pro-2:~/code2/valet-config$ kubectl get servicerole -n bookinfo
NAME                        AGE
productpage-v1-details-v1   1m
productpage-viewer          3m
[13:57:05] rick@Ricks-MacBook-Pro-2:~/code2/valet-config$ kubectl get servicerolebinding -n bookinfo
NAME                        REFERENCE                   AGE
bind-productpage-viewer     productpage-viewer          3m
productpage-v1-details-v1   productpage-v1-details-v1   1m
```

```
apiVersion: rbac.istio.io/v1alpha1
kind: ServiceRole
metadata:
  creationTimestamp: 2019-10-30T17:55:49Z
  generation: 1
  name: productpage-v1-details-v1
  namespace: bookinfo
  ownerReferences:
  - apiVersion: access.smi-spec.io/v1alpha1
    blockOwnerDeletion: true
    controller: true
    kind: TrafficTarget
    name: productpage-v1-details-v1
    uid: 81bef9a7-fb3e-11e9-867e-42010a800172
  resourceVersion: "23950"
  selfLink: /apis/rbac.istio.io/v1alpha1/namespaces/bookinfo/serviceroles/productpage-v1-details-v1
  uid: 81df089b-fb3e-11e9-867e-42010a800172
spec:
  rules:
  - constraints:
    - key: destination.user
      values:
      - details-v1
    - key: destination.namespace
      values:
      - bookinfo
    methods:
    - '*'
    paths:
    - ""
    services:
    - '*'
status: {}
```

```
apiVersion: rbac.istio.io/v1alpha1
kind: ServiceRoleBinding
metadata:
  creationTimestamp: 2019-10-30T17:55:49Z
  generation: 1
  name: productpage-v1-details-v1
  namespace: bookinfo
  ownerReferences:
  - apiVersion: access.smi-spec.io/v1alpha1
    blockOwnerDeletion: true
    controller: true
    kind: TrafficTarget
    name: productpage-v1-details-v1
    uid: 81bef9a7-fb3e-11e9-867e-42010a800172
  resourceVersion: "23951"
  selfLink: /apis/rbac.istio.io/v1alpha1/namespaces/bookinfo/servicerolebindings/productpage-v1-details-v1
  uid: 81e1c733-fb3e-11e9-867e-42010a800172
spec:
  roleRef:
    kind: ServiceRole
    name: productpage-v1-details-v1
  subjects:
  - user: cluster.local/ns/bookinfo/sa/productpage-v1
status: {}
```

This example deviated slightly from the one linked, because I had trouble getting that one to behave as expected and this was a bit simpler to understand what should be going on.

**Comment by rickducott** [comment]
Ok, so it turns out this is an istio bug -- here's the corresponding issue: https://github.com/istio/istio/pull/17430

Unfortunately that fix was only merged into master and released in 1.4.0-beta.0+, which means the istio SMI adapter doesn't work for istio versions prior to 1.4. I've validated this example works as expected with istio-1.4. Updating the summary to reflect this.

**Comment by rickducott** [comment]
Left open because it probably makes sense to clarify somewhere the istio compability
